Add auth0Client to context

kenahrens opened the pull request:
If you do not add auth0Client here then it will always return undefined if you try to use it in your own hooks.

I saw some notes in the forums that this was an issue and of course ran into it myself:
https://community.auth0.com/t/auth0client-is-undefined/27202

stevehobbsdev commented:
@kenahrens Thanks for raising. This should not be needed. Can you provide an example of your usage so I can see whether there is something else we can do?

kenahrens commented:
@stevehobbsdev I apologize for the delay in following up, I could not find the code change that I made and realized it was in a branch. The specific issue I found was that isAuthenticated and user were always coming back as undefined. I modified my code to wait for loading to be set to false and then get the values from auth0Client directly and the problem went away.

I wish I knew more about react, but I think it has something to do with the useEffect() in your sample code, specifically that the dependency list is empty:

```
  useEffect(() => {
    // most of the code removed  
    // eslint-disable-next-line
  }, []);
```

When I created my own hook I implemented it like this (I'm using react-admin project and it wants an authProvider object setup in a particular way).

```
  useEffect(function mountAuth() {

    // Can only check auth once the system is up
    if (!loading && authProvider === null) {
      const authWrapper = AuthWrapper(loading, auth0Client);
      setAuthProvider(authWrapper);
    }
  }, [authProvider, loading, auth0Client]);
```

stevehobbsdev commented:
@kenahrens We have now released a beta version of our [React SDK](https://github.com/auth0/auth0-react) which more or less supersedes the sample code here. The wrapper you're using was always designed to be replaced by a library in its own right, and now we have it.

Could you give the library a try and see if it solves your issue? If you have any feedback to give regarding that, please raise on the Auth0 React repo I just linked.

kenahrens commented:
Thank you Steve I am going to check that out.  :)
